packages/p2p/src/hooks/__tests__/use-p2p-advertiser-adverts.spec.ts

```diff
@@ -0,0 +1,151 @@
+        mockUseStores.mockReturnValue({
+            general_store: {
+                is_advertiser_info_subscribed: false,
+                counterparty_advertiser_id: 'test id 1234',
+            },
+            buy_sell_store: {
+                selected_local_currency: 'USD',
+            },
+            advertiser_page_store: { counterparty_type: 'buy' },
+        });
+        const { result } = renderHook(() => useP2PAdvertiserAdverts());
+        const adverts_list = result.current.adverts;
+
+        expect(adverts_list).toHaveLength(1);
+        expect(adverts_list?.[0]?.country).toBe('id');
+        expect(adverts_list?.[0]?.counterparty_type).toBe('buy');
+        expect(adverts_list?.[0]?.description).toBe('Created by script. Please call me 02203400');
+        expect(adverts_list?.[0]?.advertiser_details?.name).toBe('client Test90000253');
+    });
+
+    it('should return the advertiser adverts object from response when is_advertiser_info_subscribed is false and counterparty_advertiser_id is undefined', () => {
+        mockUseStores.mockReturnValue({
+            general_store: {
+                is_advertiser_info_subscribed: false,
+                counterparty_advertiser_id: undefined,
+            },
+            buy_sell_store: {
+                selected_local_currency: 'USD',
+            },
+            advertiser_page_store: { counterparty_type: 'buy' },
+        });
+        const { result } = renderHook(() => useP2PAdvertiserAdverts());
+        const adverts_list = result.current.adverts;
+
+        expect(adverts_list).toHaveLength(1);
+        expect(adverts_list?.[0]?.country).toBe('id');
+        expect(adverts_list?.[0]?.counterparty_type).toBe('buy');
+        expect(adverts_list?.[0]?.description).toBe('Created by script. Please call me 02203400');
+        expect(adverts_list?.[0]?.advertiser_details?.name).toBe('client Test90000253');
+    });
+    it('should return the advertiser adverts object from response when is_advertiser_info_subscribed is true and counterparty_advertiser_id is undefined', () => {
+        mockUseStores.mockReturnValue({
+            general_store: {
+                is_advertiser_info_subscribed: true,
+                counterparty_advertiser_id: undefined,
+            },
+            buy_sell_store: {
+                selected_local_currency: 'USD',
+            },
+            advertiser_page_store: { counterparty_type: 'buy' },
+        });
+        const { result } = renderHook(() => useP2PAdvertiserAdverts());
+        const adverts_list = result.current.adverts;
+
+        expect(adverts_list).toHaveLength(1);
+        expect(adverts_list?.[0]?.country).toBe('id');
+        expect(adverts_list?.[0]?.counterparty_type).toBe('buy');
+        expect(adverts_list?.[0]?.description).toBe('Created by script. Please call me 02203400');
```

Comment:
@jim-deriv for enhancing the maintainability of this test file in future, i suggest that we introduce a variable for this repeatedly used string :)

Reply:
Hi @maryia-deriv, This is a good suggestion, but if I introduce a constant for all of them, won't I need to do that for `'id'`, `'but'` etc?